cognite_toolkit/_cdf_tk/client/data_classes/locations.py

```diff
@@ -0,0 +1,250 @@
+from __future__ import annotations
+
+from abc import ABC
+from dataclasses import dataclass, field
+
+# We need to import Sequence from typing and not collections.abc for
+# cognite_toolkit._parameters.read_parameter_from_init_type_hints to work on Python 3.9
+# is necessary to avoid Ruff changing the import
+from typing import Any, Literal, Sequence  # noqa
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes._base import (
+    CogniteResourceList,
+    WriteableCogniteResource,
+    WriteableCogniteResourceList,
+)
+from cognite.client.data_classes.capabilities import AllScope, Capability, IDScope
+from cognite.client.data_classes.data_modeling.ids import DataModelId, ViewId
+from typing_extensions import Self
+
+
+@dataclass(frozen=True)
+class LocationFilterScene(DataModelId):
+    external_id: str
+    space: str
+
+
+@dataclass(frozen=True)
+class LocationFilterDataModel(DataModelId):
+    external_id: str
+    space: str
+    version: str
+
+
+@dataclass(frozen=True)
+class LocationFilterView(ViewId):
+    external_id: str
+    space: str
+    version: str | None = None
+    represents_entity: Literal["MAINTENANCE_ORDER", "OPERATION", "NOTIFICATION", "ASSET"] | None = None
+
+
+@dataclass(frozen=True)
+class LocationFilterAssetCentricBaseFilter:
+    data_set_ids: list[int] | None = None
+    asset_subtree_ids: list[dict[str, str]] | None = None
```

Comment:
I am in favor of introducing a small class her Identifier. 
```suggestion
    asset_subtree_ids: list[dict[Literal["id", "externalId"], int | str]] | None = None
```